File: 2962-count-subarrays-where-max-element-appears-at-least-k-times/2962-count-subarrays-where-max-element-appears-at-least-k-times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int cnt = 0, lo = 0, maxNum = 0;
        long result = 0;

        for (int num: nums) {
            maxNum = Math.max(maxNum, num);
        }

        for (int hi = 0; hi < nums.length; hi++) {
            if (nums[hi] == maxNum) {
                cnt++;
            }

            while (cnt >= k) {
                result += (long)nums.length - hi;
                if (nums[lo] == maxNum) {
                    cnt--;
                }
                lo++;
            }
        }

        return result;
    }
}

File: 2962-count-subarrays-where-max-element-appears-at-least-k-times/2962-count-subarrays-where-max-element-appears-at-least-k-times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        cnt, result, lo, maxNum = 0, 0, 0, max(nums)

        for hi in range(len(nums)):
            if nums[hi] == maxNum:
                cnt += 1

            while cnt >= k:
                result += len(nums) - hi
                if nums[lo] == maxNum:
                    cnt -= 1
                lo += 1

        return result